src/liballoc/sync.rs

```diff
@@ -49,9 +49,9 @@ const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 ///
 /// The type `Arc<T>` provides shared ownership of a value of type `T`,
 /// allocated in the heap. Invoking [`clone`][clone] on `Arc` produces
-/// a new pointer to the same value in the heap. When the last `Arc`
-/// pointer to a given value is destroyed, the pointed-to value is
-/// also destroyed.
+/// a new pointer to the same `Arc` reference value in the heap. When the last
```

Comment:
It would be more accurate to say "produces _a new `Arc` pointing to the same value_ in the heap". The [source](https://github.com/rust-lang/rust/blob/5399616f1d98b4bcc1da87af15f75c95e3c2288b/src/liballoc/sync.rs#L741) might make it clearer.

Reply:
Ooh it works like that! So the internal reference is cloned. Been hearing varying descriptions on how it works. I can change it to reflect that soon. Thanks for clarifying.